### PR TITLE
Add jdk default trust manager to KeyVaultTrustManager

### PR DESCRIPTION
**Pull request by michaelqi793**
Currently, If we override the JCA trust manager globally by KeyVaultTrustManager, when KeyVaultTrustManager is instantiated, if the passed-in key store is null, then KeyVaultTrustManager will create  a KeyVaultKeyStore then use it to initialize JDK default trust manager which then calls the KeyVaultKeyStore's getAliases() method which then needs a TSL connection that also creates new KeyVaultTrustManager... So there is a dead loop. To halt this dead loop:
1 Remove the logic of initializing the JDK default trust manager with KeyVaultKeyStore in KeyVaultTrustManager's constructor
2 Add a global static jdk trust manager initialized with jdk key store
3 Add the logic of using global static trust manager to check server/client side at the first place to block the dead loop.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @michaelqi793 , if this PR is ready to ready for review, please click `Ready for review` button.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultTrustManager.java [hunk 1]
```diff
@@ -32,10 +32,21 @@
     /**
      * Stores the default trust manager.
      */
-    private X509TrustManager defaultTrustManager;
+    private static final X509TrustManager DEFAULT_TRUST_MANAGER = defaultTrustManager();
 
     /**
-     * Stores the keystore.
+     * Stores the default trust manager with key vault key store.
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultTrustManager.java:38]: Can we just keep delete `kvTrustManager` and `otherTrustManager`, and make defaultTrustManager use the jreKeyStore?

**Comment by michaelqi793** [comment]
Replaced with #23923

**Pull request closed without merging**